File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

const appRoom = consumer.subscriptions.create("RoomChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    const msg_post_id = data['message'].match(/"\d*[0-9]"/);
    const post_id = msg_post_id[0].replace(/"/g, "");
    const messages = document.getElementById("messages"+ post_id);
    messages.insertAdjacentHTML('beforeend', data['message']);
    messages.scrollTo(0, messages.scrollHeight);
  },

  speak: function(message, post_id) {
    return this.perform('speak', {message: message, post_id: post_id});
  }
});

$(document).on("keydown", ".room_message_form_textarea",  function(e) {
  if (e.key === "Enter" && e.shiftKey === true) {
    const post_id = $('textarea').data('post_id')
    appRoom.speak(e.target.value, post_id);
    e.target.value = '';
    e.preventDefault();
  }
})

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    new_message = Message.create!(content: data['message'], user_id: current_user.id, post_id: data['post_id'])
    ActionCable.server.broadcast 'room_channel', {message: render_message(new_message)}
  end

  private


  def render_message(message)
    ApplicationController.renderer.render(partial: 'shared/message', locals: { message: message, current_user: current_user })
  end
end
